build/root/Makefile

```diff
@@ -179,6 +179,8 @@ check test:
 	@echo "$$CHECK_TEST_HELP_INFO"
 else
 check test: generated_files
+	$(eval UNKNOWN = $(shell echo $(MAKEFLAGS) | sed s/"\s"/"\n"/g | grep "=" | grep -v "^WHAT=" | grep -v "^TESTS="))
+	$(if $(UNKNOWN),$(error "$(UNKNOWN) is not supported."))
```

Comment:
```
cloud@ubuntu:~/go/src/k8s.io/kubernetes$ make test WHAAT=./pkg/api/validation/
Makefile:182: *** "WHAAT=./pkg/api/validation/ is not supported.".  Stop.
```

Awesome! It works! However, we should validate all the flags it supported. 

For example, we should allow to use `KUBE_COVER=y`.
```
cloud@ubuntu:~/go/src/k8s.io/kubernetes$ make test KUBE_COVER=y
Makefile:182: *** "KUBE_COVER=y is not supported.".  Stop.
```

Reply:
Thanks for your review, @xiangpengzhao 
That is a nice point, I will add these flags also soon.